llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -757,30 +769,31 @@ RegInterval WaitcntBrackets::getRegInterval(const MachineInstr *MI,
 
   RegInterval Result;
 
-  unsigned Reg = TRI->getEncodingValue(AMDGPU::getMCReg(Op.getReg(), *ST)) &
-                 AMDGPU::HWEncoding::REG_IDX_MASK;
+  unsigned Reg = getRegPoint(*ST, Op.getReg(), *TRI);
+  const TargetRegisterClass *RC = TRI->getPhysRegBaseClass(Op.getReg());
+  unsigned Size = TRI->getRegSizeInBits(*RC);
 
+  // VGPRs are tracked every 16 bits, SGPRs by 32 bits
```

Comment:
AGPRs are also tracked every 16 bits, right? Should mention that in the comment.

Reply:
Relatedly, we should do something about #105814

Reply:
Updated the comment.

It seems https://github.com/llvm/llvm-project/issues/105814 is about optimizing spgr intervals. I think that can be done in a seperate patch

Reply:
Just realized https://github.com/llvm/llvm-project/commit/d6c0839c9c823754fb84e47c5bacf1a3b3d0f618 is merged. Rebased again